cnd/src/btsieve/ethereum/mod.rs

```diff
@@ -222,51 +222,55 @@ where
     for transaction in block.transactions.into_iter() {
         let tx_hash = transaction.hash;
 
-        let receipt = if needs_receipt {
-            let receipt = connector
-                .receipt_by_hash(tx_hash)
-                .compat()
-                .await?
-                .ok_or_else(|| {
-                    anyhow::anyhow!(
-                        "Could not get transaction receipt for transaction {:x}",
-                        tx_hash
-                    )
-                })?;
-
-            Some(receipt)
-        } else {
-            None
-        };
-
-        let result = pattern.matches(&transaction, receipt.as_ref());
+        let receipt = connector
```

Comment:
I don't understand these changes @D4nte, with this applied it seems now that we do not consider the bloom filter results and just fetch every receipt irrespective?